doc/examples/filters/plot_blind_deconvolution.py

```diff
@@ -0,0 +1,115 @@
+"""
+=====================
+Blind Image Deconvolution
+=====================
+Normally, image deconvolution is based on a prior knowledge of the
+Point Spread Function (PSF) used to deconvolve the image.
+However, _blind_ methods are available that estimate the PSF
+iteratively from the image itself. This algorithm is based on the
+Richardson Lucy (RL) deconvolution algorithm. In this case,
+the RL algorithm is not only used for deconvolving the image,
+but also the PSF estimate. This process is iterative,
+alternating between deconvolving the PSF and deconvolving the image.
+
+The following example shows a centered cross that was convolved
+with a gaussian kernel with sigma = 2. Further Poisson
+shot noise was added. Using the convolved image as argument
+in the blind image deconvolution function, the algorithm
+is capable to recover to a large extent the original image
+and a good guess for the PSF (Figure 1).
+
+Figure 2 shows the iterative progress where the first images look
+like artifacts, but after a given amount of iterations,
+the cross and the PSF nicely show up. Figure 3 shows the absolute
+difference between the original image and the deconvolved one
+indicating that the optimal number of iterations is 33.
+
+.. [1] William Hadley Richardson, "Bayesian-Based Iterative
+       Method of Image Restoration",
+       J. Opt. Soc. Am. A 27, 1593-1607 (1972), :DOI:`10.1364/JOSA.62.000055`
+
+.. [2] https://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution
+
+.. [3] Fish, D. A., A. M. Brinicombe, E. R. Pike, and J. G. Walker.
+       "Blind deconvolution by means of the Richardson–Lucy algorithm."
+       JOSA A 12, no. 1 (1995): 58-65. :DOI:`10.1364/JOSAA.12.000058`
+
+       https://pdfs.semanticscholar.org/9e3f/a71e22caf358dbe873e9649f08c205d0c0c0.pdf
+"""
+import numpy as np
+from scipy.signal import convolve2d
+from skimage.filters import gaussian
+import matplotlib.pyplot as plt
+from skimage.restoration import richardson_lucy
+
+# Initialize image that is to be recovered (cross)
+im = np.zeros((100, 100), dtype=np.float32)
```

Comment:
We prefer to use more meaningful names for variables (at least, we tried to improve our examples this way). So, we tend to avoid im or img as image names. Can you improve that please?

Reply:
Fixed in your branch